hadoop-common-project/hadoop-common/src/main/proto/RpcHeader.proto

```diff
@@ -157,6 +158,7 @@ message RpcResponseHeaderProto {
   optional bytes clientId = 7; // Globally unique client ID
   optional sint32 retryCount = 8 [default = -1];
   optional int64 stateId = 9; // The last written Global State ID
+  map<string, int64> nameserviceStateIds = 10; // Last seen state IDs for multiple nameservices.
```

Comment:
@goiri The classes that will interact with these header fields will be in the hadoop-hdfs-client module, so it is hard to then keep changes and tests contained within hadoop-common. I don't have a workaround at the moment but I'll keep trying to think of alternative ways to test. Let me know if you have any ideas with regards to this.

Reply:
Doesn't this field need to be `optional` for backwards compatibility?

Reply:
Yes, it should be optional. Thanks for spotting the error. I'll fix that.
I'm also going to make this a bytearray since the client doesn't need to parse it. This will give us the flexibility to evolve the contents in the routers.